rclcpp_components/src/component_manager.cpp

```diff
@@ -188,6 +196,17 @@ ComponentManager::on_load_node(
       }
 
       auto options = create_node_options(request);
+      bool use_dedicated_thread = false;
+      for (const auto & a : request->extra_arguments) {
+        const rclcpp::Parameter extra_argument = rclcpp::Parameter::from_parameter_msg(a);
+        if (extra_argument.get_name() == "use_dedicated_thread") {
+          if (extra_argument.get_type() != rclcpp::ParameterType::PARAMETER_BOOL) {
+            throw ComponentManagerException(
+                    "Extra component argument 'use_dedicated_thread' must be a boolean");
+          }
+          use_dedicated_thread = extra_argument.get_value<bool>();
+        }
+      }
```

Comment:
yes, this is a duplicated code, but i think `std::map<std::string, rclcpp::Parameter>` may be a better choice to avoid multiple O(n) search, but it will consume more memory, what do you think about?

Reply:
> i think `std::map<std::string, rclcpp::Parameter>` may be a better choice to avoid multiple O(n) search, but it will consume more memory, what do you think about?

Yeah, I agree with you.